common/nym-lp/src/session.rs

```diff
@@ -6,19 +6,118 @@
 //! This module implements session management functionality, including replay protection
 //! and Noise protocol state handling.
 
+use crate::keypair::{PrivateKey, PublicKey};
 use crate::message::{EncryptedDataPayload, HandshakeData};
 use crate::noise_protocol::{NoiseError, NoiseProtocol, ReadResult};
 use crate::packet::LpHeader;
+use crate::psk::{psq_initiator_create_message, psq_responder_process_message};
 use crate::replay::ReceivingKeyCounterValidator;
 use crate::{LpError, LpMessage, LpPacket};
+use nym_crypto::asymmetric::ed25519;
+use nym_kkt::ciphersuite::{DecapsulationKey, EncapsulationKey};
 use parking_lot::Mutex;
 use snow::Builder;
-use std::sync::atomic::{AtomicU64, Ordering};
+use std::sync::atomic::{AtomicBool, AtomicU64, Ordering};
+
+/// KKT (KEM Key Transfer) exchange state.
+///
+/// Tracks the KKT protocol for obtaining the responder's KEM public key
+/// before PSQ can begin. This allows post-quantum KEM algorithms to be
+/// used even when keys are not pre-published.
+///
+/// # State Transitions
+///
+/// **Initiator path:**
+/// ```text
+/// NotStarted → InitiatorWaiting → Completed
+/// ```
+///
+/// **Responder path:**
+/// ```text
+/// NotStarted → ResponderProcessed
+/// ```
+pub enum KKTState {
```

Comment:
I wonder, would it be beneficial to have an explicit Error state?

Reply:
We immediatly do something else on errors, so we'd not be in the state long...